validate_service.rb

```diff
@@ -0,0 +1,36 @@
+require 'hoodoo'
+
+class ValidateImplementation < Hoodoo::Services::Implementation
+  def create( context )
+    body = context.request.body              # Get the body of the incoming request.
+    first_name = body['first_name']          # Get the 'first_name' value from the hash and save as variable. (json is passed into service as a string.)
```

Comment:
🐼 <- this signifies an optional change you can choose to apply or not!

No need to put `first_name` and `surname` into variables if s only used when we render the output. Instead you remove the variables and do this when rendering the output:

```
context.response.set_resource( { 'message' => "Hello #{body['first_name']} #{body['surname']}" }
```